src/DOMWrap.jsx

```diff
@@ -13,15 +17,142 @@ export default class DOMWrap extends React.Component {
     className: '',
   };
 
+  state = {
+    overflowingIndex: undefined,
+  };
+
+  overflowedItems = [];
+
+  childrenCache = [];
+
+  handleResize = () => {
+    if (this.props.mode !== 'horizontal') {
+      return;
+    }
+    // reset children to childrenCache
+    this.props.children.forEach((c, i) => this.props.children[i] = React.cloneElement(this.childrenCache[i]));
+
+    const ul = ReactDOM.findDOMNode(this);
+    const scrollWidth = ul.scrollWidth;
+    const width = ul.getBoundingClientRect().width;
+
+    this.overflowedItems = [];
+    let currentSumWidth = 0;
+    // last index of menuItem that is overflowed
+    let overflowingIndex = -1;
+    const children = this.props.children;
+
+    // try to find all the overflowed items
+    if (scrollWidth > width) {
+      let lastVisibleChild;
+      Array.prototype.slice.apply(ul.children).forEach((li, index) => {
+        const liWidth = li.getBoundingClientRect().width;
+        currentSumWidth += liWidth;
+        if (currentSumWidth > width) {
+          // somehow children[index].key is in the format of '.$key', we have to overwrite with the correct key
+          this.overflowedItems.push(React.cloneElement(children[index], { key: children[index].props.eventKey }));
+        } else {
+          overflowingIndex++;
+          lastVisibleChild = children[index];
+        }
+      });
+
+      // we need to replace last visible child with a overflow indicator ('...')
+      this.overflowedItems.unshift(React.cloneElement(lastVisibleChild, { key: lastVisibleChild.props.eventKey }));
+
+      const selectedIndex = this.childrenCache.findIndex(c => {
+        return c.props.selectedKeys.includes(c.props.eventKey);
+      });
+
+      const lastVisibleIndex = this.props.children.findIndex(c => c.props.eventKey === lastVisibleChild.props.eventKey) - 1;
+
+      // try to move the selected item out of the overflowed item
+      if (selectedIndex !== -1 && selectedIndex > lastVisibleIndex) {
+        const selectedIndexInOverflow = this.overflowedItems.findIndex(ele => ele.props.eventKey === this.childrenCache[selectedIndex].props.eventKey)
+        const tmp = this.props.children[lastVisibleIndex - 1];
+        this.props.children[lastVisibleIndex] = this.overflowedItems[selectedIndexInOverflow];
```

Comment:
Might not be the best way since we're modifying this.props.children. Suggestions appreciated!